fhir/com.b2international.snowowl.fhir.core/src/com/b2international/snowowl/fhir/core/provider/FhirApiProvider.java

```diff
@@ -91,26 +92,23 @@ protected final String getBranchPath(String version) {
 	 * @param logicalId
 	 * @return
 	 */
-	protected CodeSystemVersionEntry findCodeSystemVersion(LogicalId logicalId) {
-		
-		Optional<CodeSystemVersionEntry> codeSystemOptional = CodeSystemRequests.prepareSearchCodeSystemVersion()
+	protected CodeSystemVersion findCodeSystemVersion(LogicalId logicalId) {
+		return CodeSystemRequests.prepareSearchCodeSystemVersion()
 			.one()
 			.filterByBranchPath(logicalId.getBranchPath())
 			.build(getRepositoryId())
 			.execute(getBus())
 			.getSync()
-			.first();
-			
-		return codeSystemOptional.orElseThrow(() -> 
-			new BadRequestException(String.format("Could not find corresponding version [%s] for logical id [%s].", logicalId.getBranchPath(), logicalId), "ValueSet.id"));
+			.first()
+			.orElseThrow(() -> new BadRequestException(String.format("Could not find corresponding version [%s] for logical id [%s].", logicalId.getBranchPath(), logicalId), "ValueSet.id"));
```

Comment:
The exception message will use the output of `String.format` as a format string; I don't see where the last parameter fits into this picture.

```suggestion
			.orElseThrow(() -> new BadRequestException("Could not find corresponding version [%s] for logical id [%s].", logicalId.getBranchPath(), logicalId));
```